src/Testcontainers/Configurations/Containers/ITestcontainersConfiguration.cs

```diff
@@ -105,5 +106,14 @@ public interface ITestcontainersConfiguration : IDockerResourceConfiguration
     /// This callback will be executed after starting the container, but before executing the wait strategies.
     /// </remarks>
     Func<ITestcontainersContainer, CancellationToken, Task> StartupCallback { get; }
+
+    /// <summary>
+    /// Get the modifier callbacks for <see cref="CreateContainerParameters"/>.
+    /// </summary>
+    /// <remarks>
+    /// These actions will be executed after the remaining Testcontainer configuration has been applied on the Docker
+    /// client, but just before creating the container.
+    /// </remarks>
+    IList<Action<CreateContainerParameters>> ParameterModifiers { get; }
```

Comment:
I don't quite understand this comment?

Reply:
That's nonsense, I meant `IEnumerable<Action<CreateContainerParameters>>`

Reply:
Once again, I really don't like the idea of documenting `ParameterModifiers` as an `IEnumerable` when we can (and IMO should) make a stronger guarantee in the contract. We need to promise that the order of `ParameterModifiers` is retained, since it could have really confusing consequences if the order changes over time - and `IEnumerable` is too generic to express such a guarantee. The correct collection type here is `IList` since it expresses exactly the contract intended with `ParameterModifiers`.

But this is your project after all. If you want me to change it to `IEnumerable`, I will, but I also think it is a mistake ;)

Reply:
> The correct collection type here is `IList` since it expresses exactly the contract intended with `ParameterModifiers`

Immutable is a contract too, and this won't work well with `IList`.

> But this is your project after all.

It's a community project. My personal preferences don't matter 😃. I think both points are valid, there is just no type that covers both in .NET. Usually, I try to reduce changes to a minimal, but let's stick to `IList<T>` and add a comment?

Reply:
Ah, now I see what you were getting at with immutability. This makes perfect sence.

I am not that well versed in the dotnet collection types, but how about an `IReadOnlyList<T>` or `IImmutableList<T>`? Or do you think that gets too specific?

Reply:
> I see your point with immutability. I am not that well versed in the dotnet collection types, but how about an `IReadOnlyList<T>` or `IImmutableList<T>`?

Me neither. I checked `IImmutableList<T>` too, but that's not part of `netstandard` and `IReadOnlyCollection<T>` is just an `IEnumerable<T>`. But I missed `IReadOnlyList<T>`. That sounds right 👏.

Reply:
Perfect, I will update the PR later tonight to use `IReadOnlyList<T>` instead.